src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -207,38 +212,79 @@ public Result addFeedbackToResult(@NotNull Result result, List<Feedback> feedbac
+                results.forEach(result -> result.filterSensitiveFeedbacks(!finalShouldResultsBePublished));
             }
             else {
-                boolean applyFilter = exerciseDateService.isBeforeDueDate(result.getParticipation())
-                        || (AssessmentType.AUTOMATIC.equals(result.getAssessmentType()) && exerciseDateService.isBeforeLatestDueDate(exercise));
-                result.filterSensitiveFeedbacks(applyFilter);
+                boolean beforeLatestDueDate = exerciseDateService.isBeforeLatestDueDate(exercise);
+                boolean participationBeforeDueDate = exerciseDateService.isBeforeDueDate(participation);
+                results.forEach(result -> {
+                    boolean isBeforeDueDateOrAutomaticAndBeforeLatestDueDate = participationBeforeDueDate
+                            || (AssessmentType.AUTOMATIC.equals(result.getAssessmentType()) && beforeLatestDueDate);
+                    result.filterSensitiveFeedbacks(isBeforeDueDateOrAutomaticAndBeforeLatestDueDate);
+                });
+
+                results.forEach(result -> {
+                    boolean assessmentTypeSetAndNonAutomatic = result.getAssessmentType() != null && result.getAssessmentType() != AssessmentType.AUTOMATIC;
+                    boolean assessmentDueDateSetAndNotOver = exercise.getAssessmentDueDate() != null && ZonedDateTime.now().isBefore(exercise.getAssessmentDueDate());
+
+                    // A tutor is allowed to access all feedback, but filter for a student the manual feedback if the assessment due date is not over yet
+                    if (assessmentTypeSetAndNonAutomatic && assessmentDueDateSetAndNotOver) {
+                        // filter all non-automatic feedbacks
+                        result.setFeedbacks(result.getFeedbacks().stream().filter(feedback -> feedback.getType() != null && feedback.getType() == FeedbackType.AUTOMATIC).toList());
+                    }
+                });
             }
-            feedbacks = result.getFeedbacks();
-
-            boolean resultSetAndNonAutomatic = result.getAssessmentType() != null && result.getAssessmentType() != AssessmentType.AUTOMATIC;
-            boolean dueDateNotSetOrNotOver = exercise.getAssessmentDueDate() != null && ZonedDateTime.now().isBefore(exercise.getAssessmentDueDate());
-
-            // A tutor is allowed to access all feedback, but filter for a student the manual feedback if the assessment due date is not over yet
-            if (!exercise.isExamExercise() && resultSetAndNonAutomatic && dueDateNotSetOrNotOver) {
-                // filter all non-automatic feedbacks
-                feedbacks = feedbacks.stream().filter(feedback -> feedback.getType() != null && feedback.getType() == FeedbackType.AUTOMATIC).toList();
-            }
-        }
-
-        // remove unnecessary data to keep the json payload smaller
-        for (Feedback feedback : feedbacks) {
-            feedback.setResult(null);
         }
```

Comment:
As team scale suggests maybe add one method for exam and one for course exercises?